modules/tinymce/src/plugins/image/main/ts/core/ImageData.ts

```diff
@@ -71,8 +71,12 @@ const hasCaption = (image: HTMLElement): boolean => {
   return image.parentNode !== null && image.parentNode.nodeName === 'FIGURE';
 };
 
-const setAttrib = (image: HTMLElement, name: string, value: string) => {
-  image.setAttribute(name, value);
+const updateAttrib = (image: HTMLElement, name: string, value: string) => {
```

Comment:
Would this just be better using DOMUtils instead? As that'd allow it to handle it the same way in that it removes the attribute if you pass it null or an empty string.

Reply:
That would mean that we would have to pass around dom utils since it sets data-mce- variants depending on editor instance settings. I think I will leave it like this to not make this PR larger than it is.

Reply:
Fair enough. We have used the global DOMUtils before, which is more what I meant, but I'm fine with whatever you want to do Spocke 😄

Reply:
The global would have been wrong since I think it will then use global rules for setting data-mce-style and data-mce-src etc. This however syncs the src and deletes the data-mce-style variant so it should be good it adds some extra cpu to the parse time though.